src/abstractdataframe/join.jl

```diff
@@ -121,10 +121,10 @@ function compose_joined_table(joiner::DataFrameJoiner, kind::Symbol,
         # about the permutation of left data frame in rightjoin as we always
         # assign 0x1 to it anyway and these rows are guaranteed to come first
         # (even if they are permuted)
-        left_indicator = zeros(UInt8, nrow)
-        left_indicator[axes(all_orig_left_ixs, 1)] .= 0x1
-        right_indicator = zeros(UInt8, nrow)
-        right_indicator[axes(all_orig_right_ixs, 1)] .= 0x2
+        left_indicator = zeros(UInt32, nrow)
+        left_indicator[axes(all_orig_left_ixs, 1)] .= UInt32(1)
+        right_indicator = zeros(UInt32, nrow)
+        right_indicator[axes(all_orig_right_ixs, 1)] .= UInt32(2)
```

Comment:
```suggestion
        left_indicator[axes(all_orig_left_ixs, 1)] .= 1
        right_indicator = zeros(UInt32, nrow)
        right_indicator[axes(all_orig_right_ixs, 1)] .= 2
```

Reply:
Why change this?

`UInt32(1)` will get optimized during compilation and I thought it is more explicit.

Reply:
Does it really make a difference? I would guess LLVM will optimize it away.